lib/private/Metadata/FileEventListener.php

```
@@ -52,13 +52,26 @@ private function shouldExtractMetadata(Node $node): bool {
 		}
 
 		$path = $node->getPath();
+		return $this->isCorrectPath($path);
+	}
+
+	private function isCorrectPath(string $path): bool {
 		// TODO make this more dynamic, we have the same issue in other places
 		return !str_starts_with($path, 'appdata_') && !str_starts_with($path, 'files_versions/') && !str_starts_with($path, 'files_trashbin/');
 	}
 
 	public function handle(Event $event): void {
 		if ($event instanceof NodeRemovedFromCache) {
+			if (!$this->isCorrectPath($event->getPath())) {
+				// Don't listen to paths for which we don't extract metadata
+				return;
+			}
 			$view = Filesystem::getView();
+			if (!$view) {
+				// Should not happen since a scan in the user folder should setup
+				// the file system.
```

If this is not supposed to happen, I would throw an exception here instead of returning.

---

Not sure this is a good idea to throw an error as this would abord the scan but I should definitively at least log it as an error.